### ISSUE TITLE
Attribute incorrectly added for opening JSX tag

### ISSUE THREAD

**Issue by ericsj** [microsoft/TypeScript#47090]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.63.0
- OS Version: Ubuntu 20.04.1

Steps to Reproduce:

1. Create a functional React component
2. Inside the element's return create a React Fragment
3. Inside the fragment's scope type "<" and start typing a html element name until intellisense suggests some. Type tab or enter. Instead of "button" for example, it will place "button={}"

**Comment by mjbvz** [comment]
@armanio123 @RyanCavanaugh This looks like a potential candidate for the TS recovery build. Even when you set `"javascript.preferences.jsxAttributeCompletionStyle": "none"`, the `=` still ends up being inserted

**Comment by RyanCavanaugh** [comment]
@mjbvz good catch, this is annoying as heck

**Comment by RyanCavanaugh** [comment]
```ts
namespace JSX {
    export interface IntrinsicElements {
        button: any;
        div: any;
    }
}
function fn() {
    return <>
        <butto
    </>;
}
```
Bring up completion and tab-complete at `butto`, the new text is
```
    <button={}
```
when it should be (as was in 4.4)
```
    <button
```

**Comment by jakebailey** [comment]
> Even when you set "javascript.preferences.jsxAttributeCompletionStyle": "none", the = still ends up being inserted

I can't reproduce that bit; are you in a `tsx`, which instead works on `"typescript.preferences.jsxAttributeCompletionStyle"` (note the "typescript")? The expression does check that explicitly.

In any case, I've sent #47096; what was happening was that in this broken AST, it looks like an attribute, when in fact it's the tag name. 4.4 didn't have JSX attribute snippets, so never had to care about this detail.

**Comment by mjbvz** [comment]
Thanks for taking a look @jakebailey! The VS Code recovery build will go our Tuesday. Would it be possible to get this into a published TS 4.5.4 by then?

**Comment by DanielRosenwasser** [comment]
Yes, I think we can do a cherry-pick/build.
